tensorflow/compiler/xla/mlir_hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -8871,6 +8875,47 @@ LogicalResult UniformDequantizeOp::inferReturnTypeComponents(
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// DynamicConvOp
+//===----------------------------------------------------------------------===//
+
+// simplify dynamic_conv = > convolution
+static LogicalResult simplifyDynamicConv(DynamicConvOp op,
+                                         PatternRewriter &rewriter) {
+  DenseIntElementsAttr dPaddingAttr;
+  if (!matchPattern(op.getDPadding(), m_Constant(&dPaddingAttr))) {
+    return failure();
+  }
+  size_t spatialDim = op.dimension_numbers().getInputSpatialDimensions().size();
+  assert(dPaddingAttr.size() == spatialDim * 2);
```

Comment:
Typically we prefer emitting op matcher errors and returning failure. Assertions are problematic as they don't trigger during release builds.